### ISSUE TITLE
Aligning Work against CMS Objectives

### ISSUE THREAD

**Issue by productmike** [department-of-veterans-affairs/va.gov-cms#14024]
## User Story / Task
In preparation for work to align with updated CMS objectives and outcomes, we want to (re)evaluate:

(1) The components under DevSecOps ownership. And if we believe that list needs updated (+/-).
(2) Existing OR new tickets, epics that would be pertinent to any of the 4 objectives found here: 

*OKRs:* https://app.mural.co/t/departmentofveteransaffairs9999/m/departmentofveteransaffairs9999/1684785506749/8d66873a7fb0a6c98c6ab574a0fe4f91c12f36e3?sender=ub3a2c2a4de441f561d452324

## Acceptance Criteria
- [ ] An artifact/diagram has been created & documented here that clearly identified the systems and components of the CMS environment under the DevSecOps management.
- [ ] Any existing epics (or one-off tickets) have been documented in this ticket. 
- [ ] Review those artifacts with PM. Organize for prioritization with PO.

**Comment by productmike** [comment]
@olivereri Assuming 8 story points for the purposes of planning. Feel free to update if you feel otherwise based on the ACs.

**Comment by olivereri** [comment]
![20230321_ VA Gov Platform AWS Application Infrastructure for CMS - Architecture Overview](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/31904439/c1ba0fa1-a526-4bfb-9168-a9d3d7cf1490)

**Comment by olivereri** [comment]
Pervious DevSecOps initiatives/objectives:

**Usability:** Reducing tedious tasks/toil, lowering wait times (merge to fail/success times), and reduce release related failures. Ease of CI use.

**Reliability, Availability and Stability:** Protect Production uptime, reduce deployment failures, and fault tolerance

**Observability:** Better ability to actively debug errors/failures. Awareness of user issues before it is raised. Ability to "observe" and react to novel issues or failures.

**Comment by ndouglas** [comment]
I think that looks great.

We might consider adding Innovation/Research so we can prioritize the occasional R&D ticket. IDK about you but I spend zero time outside work thinking about how to devops Drupal better. If I start thinking about it, I try to stop. So it might be worth having a ticket or two every once in a while to check our alignment/consider options, etc while on the clock.

I think we should move the Tugboat environment into code. That's sort of implicit in what you've mentioned, but it might be worth adding a bucket for Compliance/Governance/Best Practices or something like that? Plus the additional work we pick up randomly because someone says our server doesn't have HSTS turned on or our DNS service isn't hardened enough or whatever. And recurring maintenance from cert updating/changing.

Dependabot does good things for our dependencies and security, but there might be tools available to make sure we're not introducing XSS/etc vulnerabilities ourselves 😑

**Comment by olivereri** [comment]
I wish I could make more time outside of work to think about how to DevOps Drupal better. Even just basically learning more about Drupal to expand what I can do operations-wise. Making that space during work hours really is my/our only option. I know before we settled on these initiatives or objectives we had been creating epics around specific improvements.  I suppose that could fit under Usability and make sure to delineate between Operator and Developer usability improvements.

Getting Tugboat into code is going to be challenge considering the BRD system is being deprecated. A couple options I can think of:
- Reachout to Lullabot/James Snasbury and ask if Tugboat has plans to port to EKS.
- Recreate only the pieces of BRD we need to implement it in infra code. Replace Jenkins with Github Actions.

We did have a previous effort to try and put Tugboat into infra-code. It was canned in favor of porting it to Platform Op's ArgoCD/EKS system. I don't think any time was put in to how that'd work though. The epic's name is a little confusing, It was renamed after the fact. This was an effort to add Tugboat to BRD:

https://github.com/department-of-veterans-affairs/va.gov-cms/issues/3314

We really do need a specific iniative/objective that covers security related work. We are now labeling ourselves as Dev**Sec**Ops after all. Need to spend more time thinking about how to organize our Objectives.
